src/route-handlers/update-user-by-id.js

```diff
@@ -0,0 +1,38 @@
+const boom = require('boom')
+const cleanseUserForResponse = require('../utils/transform-user-record-for-response')
+
+/**
+ * handler for fetching user by ID 
+ * @param  {Hapi: Request} request        
+ * @param  {Hapi: Response} responseHandler
+ * @return {User} Fetched user
+ */
+module.exports = async function updateUserById(request, responseHandler) {
+  try {
+    const userCollection = request.mongo.db.collection('users')
+    const fetchedUser    = await userCollection.findOne({ _id: request.params.userId })
+
+    if (!fetchedUser) {
+      return boom.notFound('User with provided id was not found.')
+    } else {
+
+      const updatedUserRecord = Object.assign(request.payload, {
```

Comment:
Yea, that's a very good point. It shouldn't be written that way at all. I would correct that everywhere I used `Object.assign`.

This can also introduce bugs as its easy to overlook that request.payload is being modified. Better to lean towards functional-paradigm of not changing state.